### ISSUE TITLE
Increase validation speed during training.

### ISSUE THREAD

**Issue by jakubsadel** [open-mmlab/mmsegmentation#3363]
Hello, I have a question about the validation process after appropriate intervals/episodes during training. In my case, it takes a lot more time to go through the validation set of 215 images, compared to the duration of one epoch i.e. to go through the training set of 880 images. 

[>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>> ] 178/215, 0.2 task/s, elapsed: 1183s, ETA: 246s

 It seems to me that part of this may be due to the fact that the training batch_size aka samples_per_gpu can be set to any value, while within the validation it seems to be set to 1 by default and I don't see where I can set it.
Please advise me how i can increase speed of validation and if there is something related to these sapmples_per_gpu or maybe the problem lies somewhere else entirely. Below I leave my configuration file.
Thank you in advance :)

<details> 
  <summary>Config </summary>
 `
2023-10-05 11:57:57,039 - mmseg - INFO - Multi-processing start method is `None`
2023-10-05 11:57:57,040 - mmseg - INFO - OpenCV num_threads is `16
2023-10-05 11:57:57,295 - mmseg - INFO - Environment info:
------------------------------------------------------------
sys.platform: linux
Python: 3.8.12 (default, Sep  7 2023, 23:28:17) [GCC 7.5.0]
CUDA available: True
GPU 0,1: Tesla T4
CUDA_HOME: /usr
NVCC: Cuda compilation tools, release 9.1, V9.1.8
GCC: gcc (Ubuntu 7.5.0-3ubuntu1~18.04) 7.5.0
PyTorch: 1.7.0+cu92
PyTorch compiling details: PyTorch built with:
  - GCC 7.3
  - C++ Version: 201402
  - Intel(R) Math Kernel Library Version 2020.0.0 Product Build 20191122 for Intel(R) 64 architecture applications
  - Intel(R) MKL-DNN v1.6.0 (Git Hash 5ef631a030a6f73131c77892041042805a06064f)
  - OpenMP 201511 (a.k.a. OpenMP 4.5)
  - NNPACK is enabled
  - CPU capability usage: AVX2
  - CUDA Runtime 9.2
  - NVCC architecture flags: -gencode;arch=compute_37,code=sm_37;-gencode;arch=compute_50,code=sm_50;-gencode;arch=compute_60,code=sm_60;-gencode;arch=compute_70,code=sm_70
  - CuDNN 7.6.3
  - Magma 2.5.2
  - Build settings: BLAS=MKL, BUILD_TYPE=Release, CXX_FLAGS= -Wno-deprecated -fvisibility-inlines-hidden -DUSE_PTHREADPOOL -fopenmp -DNDEBUG -DUSE_FBGEMM -DUSE_QNNPACK -DUSE_PYTORCH_QNNPACK -DUSE_XNNPACK -DUSE_VULKAN_WRAPPER -O2 -fPIC -Wno-narrowing -Wall -Wextra -Werror=return-type -Wno-missing-field-initializers -Wno-type-limits -Wno-array-bounds -Wno-unknown-pragmas -Wno-sign-compare -Wno-unused-parameter -Wno-unused-variable -Wno-unused-function -Wno-unused-result -Wno-unused-local-typedefs -Wno-strict-overflow -Wno-strict-aliasing -Wno-error=deprecated-declarations -Wno-stringop-overflow -Wno-psabi -Wno-error=pedantic -Wno-error=redundant-decls -Wno-error=old-style-cast -fdiagnostics-color=always -faligned-new -Wno-unused-but-set-variable -Wno-maybe-uninitialized -fno-math-errno -fno-trapping-math -Werror=format -Wno-stringop-overflow, PERF_WITH_AVX=1, PERF_WITH_AVX2=1, PERF_WITH_AVX512=1, USE_CUDA=ON, USE_EXCEPTION_PTR=1, USE_GFLAGS=OFF, USE_GLOG=OFF, USE_MKL=ON, USE_MKLDNN=ON, USE_MPI=OFF, USE_NCCL=ON, USE_NNPACK=ON, USE_OPENMP=ON, 

TorchVision: 0.8.0
OpenCV: 4.8.0
MMCV: 1.6.2
MMCV Compiler: GCC 7.3
MMCV CUDA Compiler: 9.2
MMSegmentation: 0.30.0+9cdb612
------------------------------------------------------------

2023-10-05 11:57:57,296 - mmseg - INFO - Distributed training: True
2023-10-05 11:57:58,286 - mmseg - INFO - Config:
custom_imports = dict(imports=['geospatial_fm'])
dist_params = dict(backend='nccl')
log_level = 'INFO'
load_from = None
resume_from = None
cudnn_benchmark = True
dataset_type = 'GeospatialDataset'
data_root = '/home/jsadel/hls-foundation-os/data/dynamic_earth_net_fm'
num_frames = 1
img_size = 224
num_workers = 2
samples_per_gpu = 4
img_norm_cfg = dict(
    means=[
        1362.3309898376465, 1790.8744201660156, 2423.8823137283325,
        2414.259620666504, 772.4646606445312, 2389.885033607483
    ],
    stds=[
        486.5775608091387, 665.9590255041255, 860.4737781152131,
        846.6299310993978, 213.6570929844219, 815.553590122155
    ])
bands = [1, 2, 3, 8, 10, 11]
tile_size = 224
orig_nsize = 1024
crop_size = (224, 224)
img_dir = '/home/jsadel/hls-foundation-os/data/dynamic_earth_net_fm/sentinel2'
ann_dir = '/home/jsadel/hls-foundation-os/data/dynamic_earth_net_fm/labels_int16'
img_suffix = '_Sentinel2.tif'
seg_map_suffix = '_Label.tif'
ignore_index = -1
image_nodata = -9999
image_nodata_replace = 0
image_to_float32 = True
splits = dict(
    train='/home/jsadel/hls-foundation-os/data_splits/deforestation/train.txt',
    val='/home/jsadel/hls-foundation-os/data_splits/deforestation/val.txt',
    test='/home/jsadel/hls-foundation-os/data_splits/deforestation/test.txt')
pretrained_weights_path = '/home/jsadel/hls-foundation-os/Prithvi_100M.pt'
num_layers = 12
patch_size = 16
embed_dim = 768
num_heads = 12
tubelet_size = 1
output_embed_dim = 768
epochs = 10
eval_epoch_interval = 2
experiment = 'norm'
project_dir = 'deforestation'
work_dir = 'deforestation/norm'
save_path = 'deforestation/norm'
train_pipeline = [
    dict(
        type='LoadGeospatialImageFromFile',
        to_float32=True,
        channels_last=True),
    dict(type='LoadGeospatialAnnotations', reduce_zero_label=False),
    dict(type='BandsExtract', bands=[1, 2, 3, 8, 10, 11]),
    dict(type='RandomFlip', prob=0.5),
    dict(type='ToTensor', keys=['img', 'gt_semantic_seg']),
    dict(type='TorchPermute', keys=['img'], order=(2, 0, 1)),
    dict(
        type='TorchNormalize',
        means=[
            1362.3309898376465, 1790.8744201660156, 2423.8823137283325,
            2414.259620666504, 772.4646606445312, 2389.885033607483
        ],
        stds=[
            486.5775608091387, 665.9590255041255, 860.4737781152131,
            846.6299310993978, 213.6570929844219, 815.553590122155
        ]),
    dict(type='TorchRandomCrop', crop_size=(224, 224)),
    dict(type='Reshape', keys=['img'], new_shape=(6, 1, 224, 224)),
    dict(type='Reshape', keys=['gt_semantic_seg'], new_shape=(1, 224, 224)),
    dict(
        type='CastTensor',
        keys=['gt_semantic_seg'],
        new_type='torch.LongTensor'),
    dict(type='Collect', keys=['img', 'gt_semantic_seg'])
]
test_pipeline = [
    dict(
        type='LoadGeospatialImageFromFile',
        to_float32=True,
        channels_last=True),
    dict(type='BandsExtract', bands=[1, 2, 3, 8, 10, 11]),
    dict(type='ToTensor', keys=['img']),
    dict(type='TorchPermute', keys=['img'], order=(2, 0, 1)),
    dict(
        type='TorchNormalize',
        means=[
            1362.3309898376465, 1790.8744201660156, 2423.8823137283325,
            2414.259620666504, 772.4646606445312, 2389.885033607483
        ],
        stds=[
            486.5775608091387, 665.9590255041255, 860.4737781152131,
            846.6299310993978, 213.6570929844219, 815.553590122155
        ]),
    dict(
        type='Reshape',
        keys=['img'],
        new_shape=(6, 1, -1, -1),
        look_up=dict({
            '2': 1,
            '3': 2
        })),
    dict(type='CastTensor', keys=['img'], new_type='torch.FloatTensor'),
    dict(
        type='CollectTestList',
        keys=['img'],
        meta_keys=[
            'img_info', 'seg_fields', 'img_prefix', 'seg_prefix', 'filename',
            'ori_filename', 'img', 'img_shape', 'ori_shape', 'pad_shape',
            'scale_factor', 'img_norm_cfg'
        ])
]
CLASSES = ('No forest', 'Forest')
data = dict(
    samples_per_gpu=4,
    workers_per_gpu=2,
    train=dict(
        type='GeospatialDataset',
        CLASSES=('No forest', 'Forest'),
        data_root='/home/jsadel/hls-foundation-os/data/dynamic_earth_net_fm',
        img_dir=
        '/home/jsadel/hls-foundation-os/data/dynamic_earth_net_fm/sentinel2',
        ann_dir=
        '/home/jsadel/hls-foundation-os/data/dynamic_earth_net_fm/labels_int16',
        img_suffix='_Sentinel2.tif',
        seg_map_suffix='_Label.tif',
        pipeline=[
            dict(
                type='LoadGeospatialImageFromFile',
                to_float32=True,
                channels_last=True),
            dict(type='LoadGeospatialAnnotations', reduce_zero_label=False),
            dict(type='BandsExtract', bands=[1, 2, 3, 8, 10, 11]),
            dict(type='RandomFlip', prob=0.5),
            dict(type='ToTensor', keys=['img', 'gt_semantic_seg']),
            dict(type='TorchPermute', keys=['img'], order=(2, 0, 1)),
            dict(
                type='TorchNormalize',
                means=[
                    1362.3309898376465, 1790.8744201660156, 2423.8823137283325,
                    2414.259620666504, 772.4646606445312, 2389.885033607483
                ],
                stds=[
                    486.5775608091387, 665.9590255041255, 860.4737781152131,
                    846.6299310993978, 213.6570929844219, 815.553590122155
                ]),
            dict(type='TorchRandomCrop', crop_size=(224, 224)),
            dict(type='Reshape', keys=['img'], new_shape=(6, 1, 224, 224)),
            dict(
                type='Reshape',
                keys=['gt_semantic_seg'],
                new_shape=(1, 224, 224)),
            dict(
                type='CastTensor',
                keys=['gt_semantic_seg'],
                new_type='torch.LongTensor'),
            dict(type='Collect', keys=['img', 'gt_semantic_seg'])
        ],
        split=
        '/home/jsadel/hls-foundation-os/data_splits/deforestation/train.txt',
        ignore_index=-1),
    val=dict(
        type='GeospatialDataset',
        CLASSES=('No forest', 'Forest'),
        data_root='/home/jsadel/hls-foundation-os/data/dynamic_earth_net_fm',
        img_dir=
        '/home/jsadel/hls-foundation-os/data/dynamic_earth_net_fm/sentinel2',
        ann_dir=
        '/home/jsadel/hls-foundation-os/data/dynamic_earth_net_fm/labels_int16',
        img_suffix='_Sentinel2.tif',
        seg_map_suffix='_Label.tif',
        pipeline=[
            dict(
                type='LoadGeospatialImageFromFile',
                to_float32=True,
                channels_last=True),
            dict(type='BandsExtract', bands=[1, 2, 3, 8, 10, 11]),
            dict(type='ToTensor', keys=['img']),
            dict(type='TorchPermute', keys=['img'], order=(2, 0, 1)),
            dict(
                type='TorchNormalize',
                means=[
                    1362.3309898376465, 1790.8744201660156, 2423.8823137283325,
                    2414.259620666504, 772.4646606445312, 2389.885033607483
                ],
                stds=[
                    486.5775608091387, 665.9590255041255, 860.4737781152131,
                    846.6299310993978, 213.6570929844219, 815.553590122155
                ]),
            dict(
                type='Reshape',
                keys=['img'],
                new_shape=(6, 1, -1, -1),
                look_up=dict({
                    '2': 1,
                    '3': 2
                })),
            dict(
                type='CastTensor', keys=['img'], new_type='torch.FloatTensor'),
            dict(
                type='CollectTestList',
                keys=['img'],
                meta_keys=[
                    'img_info', 'seg_fields', 'img_prefix', 'seg_prefix',
                    'filename', 'ori_filename', 'img', 'img_shape',
                    'ori_shape', 'pad_shape', 'scale_factor', 'img_norm_cfg'
                ])
        ],
        split=
        '/home/jsadel/hls-foundation-os/data_splits/deforestation/val.txt',
        ignore_index=-1),
    test=dict(
        type='GeospatialDataset',
        CLASSES=('No forest', 'Forest'),
        data_root='/home/jsadel/hls-foundation-os/data/dynamic_earth_net_fm',
        img_dir=
        '/home/jsadel/hls-foundation-os/data/dynamic_earth_net_fm/sentinel2',
        ann_dir=
        '/home/jsadel/hls-foundation-os/data/dynamic_earth_net_fm/labels_int16',
        img_suffix='_Sentinel2.tif',
        seg_map_suffix='_Label.tif',
        pipeline=[
            dict(
                type='LoadGeospatialImageFromFile',
                to_float32=True,
                channels_last=True),
            dict(type='BandsExtract', bands=[1, 2, 3, 8, 10, 11]),
            dict(type='ToTensor', keys=['img']),
            dict(type='TorchPermute', keys=['img'], order=(2, 0, 1)),
            dict(
                type='TorchNormalize',
                means=[
                    1362.3309898376465, 1790.8744201660156, 2423.8823137283325,
                    2414.259620666504, 772.4646606445312, 2389.885033607483
                ],
                stds=[
                    486.5775608091387, 665.9590255041255, 860.4737781152131,
                    846.6299310993978, 213.6570929844219, 815.553590122155
                ]),
            dict(
                type='Reshape',
                keys=['img'],
                new_shape=(6, 1, -1, -1),
                look_up=dict({
                    '2': 1,
                    '3': 2
                })),
            dict(
                type='CastTensor', keys=['img'], new_type='torch.FloatTensor'),
            dict(
                type='CollectTestList',
                keys=['img'],
                meta_keys=[
                    'img_info', 'seg_fields', 'img_prefix', 'seg_prefix',
                    'filename', 'ori_filename', 'img', 'img_shape',
                    'ori_shape', 'pad_shape', 'scale_factor', 'img_norm_cfg'
                ])
        ],
        split=
        '/home/jsadel/hls-foundation-os/data_splits/deforestation/test.txt',
        ignore_index=-1))
optimizer = dict(type='Adam', lr=1.3e-05, betas=(0.9, 0.999))
optimizer_config = dict(grad_clip=None)
lr_config = dict(
    policy='poly',
    warmup='linear',
    warmup_iters=1500,
    warmup_ratio=1e-06,
    power=1.0,
    min_lr=0.0,
    by_epoch=False)
log_config = dict(
    interval=10,
    hooks=[
        dict(type='TextLoggerHook', by_epoch=True),
        dict(type='TensorboardLoggerHook', by_epoch=True)
    ])
checkpoint_config = dict(
    by_epoch=True, interval=10, out_dir='deforestation/norm')
evaluation = dict(
    interval=2, metric='mIoU', pre_eval=True, save_best='mIoU', by_epoch=True)
loss_func = dict(
    type='DiceLoss', use_sigmoid=False, loss_weight=1, ignore_index=-1)
runner = dict(type='EpochBasedRunner', max_epochs=10)
workflow = [('train', 1), ('val', 1)]
norm_cfg = dict(type='BN', requires_grad=True)
model = dict(
    type='TemporalEncoderDecoder',
    frozen_backbone=False,
    backbone=dict(
        type='TemporalViTEncoder',
        pretrained='/home/jsadel/hls-foundation-os/Prithvi_100M.pt',
        img_size=224,
        patch_size=16,
        num_frames=1,
        tubelet_size=1,
        in_chans=6,
        embed_dim=768,
        depth=12,
        num_heads=12,
        mlp_ratio=4.0,
        norm_pix_loss=False),
    neck=dict(
        type='ConvTransformerTokensToEmbeddingNeck',
        embed_dim=768,
        output_embed_dim=768,
        drop_cls_token=True,
        Hp=14,
        Wp=14),
    decode_head=dict(
        num_classes=2,
        in_channels=768,
        type='FCNHead',
        in_index=-1,
        channels=256,
        num_convs=1,
        concat_input=False,
        dropout_ratio=0.1,
        norm_cfg=dict(type='BN', requires_grad=True),
        align_corners=False,
        loss_decode=dict(
            type='DiceLoss', use_sigmoid=False, loss_weight=1,
            ignore_index=-1)),
    auxiliary_head=dict(
        num_classes=2,
        in_channels=768,
        type='FCNHead',
        in_index=-1,
        channels=256,
        num_convs=2,
        concat_input=False,
        dropout_ratio=0.1,
        norm_cfg=dict(type='BN', requires_grad=True),
        align_corners=False,
        loss_decode=dict(
            type='DiceLoss', use_sigmoid=False, loss_weight=1,
            ignore_index=-1)),
    train_cfg=dict(),
    test_cfg=dict(mode='slide', stride=(112, 112), crop_size=(224, 224)))
gpu_ids = range(0, 1)
auto_resume = False

2023-10-05 11:57:58,287 - mmseg - INFO - Set random seed to 1222185294, deterministic: False
2023-10-05 11:57:59,617 - mmseg - INFO - initialize FCNHead with init_cfg {'type': 'Normal', 'std': 0.01, 'override': {'name': 'conv_seg'}}
2023-10-05 11:57:59,618 - mmseg - INFO - initialize FCNHead with init_cfg {'type': 'Normal', 'std': 0.01, 'override': {'name': 'conv_seg'}}
Name of parameter - Initialization information
`
</details>

**Comment by QChhh123** [comment]
I have the same problem as you, have you solved it

**Comment by xinyingxy** [comment]
I also have the same problem as you
